sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/FluxAutoRenew.java

```diff
@@ -0,0 +1,164 @@
+            "'messageLockContainer' cannot be null.");
+
+        Objects.requireNonNull(maxAutoLockRenewDuration, "'maxAutoLockRenewDuration' cannot be null.");
+        if (maxAutoLockRenewDuration.isNegative() || maxAutoLockRenewDuration.isZero()) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                "'maxLockRenewalDuration' should not be zero or negative."));
+        }
+        this.maxAutoLockRenewal = maxAutoLockRenewDuration;
+
+    }
+
+    @Override
+    public void subscribe(CoreSubscriber<? super ServiceBusReceivedMessage> actual) {
+        LockRenewSubscriber newLockRenewSubscriber = new LockRenewSubscriber(actual, maxAutoLockRenewal,
+            messageLockContainer, onRenewLock);
+
+        source.subscribe(newLockRenewSubscriber);
+
+    }
+
+    /**
+     * Receives messages from to upstream, pushes them downstream and start lock renewal.
+     */
+    final class LockRenewSubscriber extends BaseSubscriber<ServiceBusReceivedMessage> {
+        private final ClientLogger logger = new ClientLogger(LockRenewSubscriber.class);
+
+        private final Function<String, Mono<OffsetDateTime>> onRenewLock;
+        private final Duration maxAutoLockRenewal;
+        private final LockContainer<LockRenewalOperation> messageLockContainer;
+        private final CoreSubscriber<? super ServiceBusReceivedMessage> actual;
+
+        LockRenewSubscriber(CoreSubscriber<? super ServiceBusReceivedMessage> actual, Duration maxAutoLockRenewDuration,
+            LockContainer<LockRenewalOperation> messageLockContainer,
+            Function<String, Mono<OffsetDateTime>> onRenewLock) {
+            this.onRenewLock = Objects.requireNonNull(onRenewLock, "'onRenewLock' cannot be null.");
+            this.actual = actual;
+            this.messageLockContainer = Objects.requireNonNull(messageLockContainer,
+                "'messageLockContainer' cannot be null.");
+            Objects.requireNonNull(maxAutoLockRenewDuration, "'maxAutoLockRenewDuration' cannot be null.");
+            this.maxAutoLockRenewal = maxAutoLockRenewDuration;
+        }
+        /**
+         * On an initial subscription, will take the first work item, and request that amount of work for it.
+         * @param subscription Subscription for upstream.
+         */
+        @Override
+        protected void hookOnSubscribe(Subscription subscription) {
+            Objects.requireNonNull(subscription, "'subscription' cannot be null.");
+            actual.onSubscribe(subscription);
+        }
+
+        /**
+         * When upstream has completed emitting messages.
+         */
+        @Override
+        public void hookOnComplete() {
+            logger.info("Upstream has completed.");
+            actual.onComplete();
+        }
+
+        @Override
+        protected void hookOnError(Throwable throwable) {
+            logger.info("Errors occurred upstream.", throwable);
```

Comment:
Informational?

Also, the underlying CoreSubscriber should do the dispose already. Do you need to explicitly do it?